core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala

```diff
@@ -239,7 +246,8 @@ private[netty] class NettyRpcEnv(
           (client, response) => onSuccess(deserialize[Any](client, response)))
         postToOutbox(message.receiver, rpcMessage)
         promise.future.failed.foreach {
-          case _: TimeoutException => rpcMessage.onTimeout()
+          case e: TimeoutException => rpcMessage.onTimeout()
+          case e: RpcCanceledException => rpcMessage.onCancel()
```

Comment:
@xuanyuanking Here I close the RPC connection.

Reply:
Thanks for marking this.
Nit: `case _: TimeoutException`